openqabot/types/incident.py

```diff
@@ -144,3 +148,53 @@ def contains_package(self, requires: List[str]) -> bool:
                 if package.startswith(req) and package != "kernel-livepatch-tools":
                     return True
         return False
+
+    def has_failures(self, token) -> bool:
+        results = requests.get(
+            QEM_DASHBOARD + "/api/jobs/incident/" + str(self.id), headers=token
+        ).json()
+
+        failed_jobs = self.filter_failures(results)
+        if failed_jobs:
+            log.info("Found %s failed jobs for incident %s:", len(failed_jobs), self.id)
+            if log.isEnabledFor(LOG_DEBUG_LEVEL):
+                list(
+                    map(lambda job: self.log_debug_incident(job["job_id"]), failed_jobs)
+                )
+
+        if not results:
+            raise NoResultsError(
+                "Job setting %s not found for incident" % (str(self.id))
+            )
+
+        # we only need to check if there are any failed jobs
+        return any(failed_jobs)
+
+    def log_debug_incident(self, job_id):
+        log.debug(
+            "Job %s is not marked as acceptable for incident %s",
+            job_id,
+            self.id,
+        )
+
+    def filter_failures(self, results):
+        return [
+            res
+            for res in results
+            if res["status"] not in ("passed", "softfailed")
```

Comment:
Are you sure the status can ever be soft failed here? I don't think this is the direct openQA job result here

Reply:
Better safe than sorry. This is a result comming directly from smelt. One important thing of this code, is that it makes clear to whoever is looking at it, what are the conditions that need to be met. And I don't see a future where we would like to give a different treatment to softfails.

Reply:
Had a look into the bot/dashboard code.
Seems like we are using  [dashboard data](https://github.com/openSUSE/qem-bot/pull/154/files/b194cd734e1da3cfe79a033a705aab6340a5a797#diff-2a7060f0a42828a4dbbbebdd465bf429c4c2d0e4d4760342da54546f1fd97480R154), and the dashboard [hides softfails as passed](https://github.com/openSUSE/qem-dashboard/blob/2d5dfece9fca7e7fc9f4c3970c096a57cbb44a8c/lib/Dashboard/Model/Jobs.pm#L103). 
So right now, if the dashboard do not change behavior, softfails will never happen in this code so we could skip it.
But that would mean building yet one more thing that only works on top of some "processed" data from the dashboard database and will break in weird ways if you try to run it with data direct from source.

Reply:
> Had a look into the bot/dashboard code. Seems like we are using [dashboard data](https://github.com/openSUSE/qem-bot/pull/154/files/b194cd734e1da3cfe79a033a705aab6340a5a797#diff-2a7060f0a42828a4dbbbebdd465bf429c4c2d0e4d4760342da54546f1fd97480R154), and the dashboard [hides softfails as passed](https://github.com/openSUSE/qem-dashboard/blob/2d5dfece9fca7e7fc9f4c3970c096a57cbb44a8c/lib/Dashboard/Model/Jobs.pm#L103). So right now, if the dashboard do not change behavior, softfails will never happen in this code so we could skip it. But that would mean building yet one more thing that only works on top of some "processed" data from the dashboard database and will break in weird ways if you try to run it with data direct from source.

I wonder if it makes more sense to query openQA directly, since that's being done too when looking for the comments. (would be done in a separate PR though)

Reply:
Yes, querying live results can be considered but please leave that to a separate PR. And regarding "softfailed", same as @michaelgrifalconi explained, does not make sense here. The "passed" is not the openQA-"passed". And "better safe than sorry" is not a good reasoning to introduce wrong and confusing code. We don't want to have

```suggestion
            if res["status"] not in ("passed", "softfailed", "not-ok", "whoknows", "whatever")
```

"just to be safe", right?

Reply:
Let's keep it technical, shall we? ;-) The dashboard can return passed, waiting, stopped or failed. If the concern here is that the dashboard might change behavior we should ensure it has according unit tests.

Reply:
FYI I filed https://github.com/openSUSE/qem-dashboard/issues/810

Reply:
> Yes, querying live results can be considered but please leave that to a separate PR. And regarding "softfailed", same as @michaelgrifalconi explained, does not make sense here. The "passed" is not the openQA-"passed". And "better safe than sorry" is not a good reasoning to introduce wrong and confusing code. We don't want to have
> 
> "just to be safe", right?

Addressed in 62cb3d29d710c849c83effa4680ae2740a1800b0, so for now let's be exclusive